joomla/joomla-cms
# [system messages] Add new options for custom title and/or hide title

#### Summary of Changes

The system messages are somewhat limited, since we can't change their title or even remove their title.

To solve this, this PR adds new options for system messages.
- Allows to hide the message title.
- Allows to add a custom title.

This PR also applies the changes need in all layouts (system, isis, protostar and beez3).

#### Testing Instructions

- Test if messages are working fine
- Then add the following code to the beggining of the `index.php` file of the Joomla templates (protostar, beez3, isis and hathor):
```php
JFactory::getApplication()->enqueueMessage('Test 1: Message without group (fallbacks to <code>message</code> group).');
JFactory::getApplication()->enqueueMessage('Test 2: Message in <code>error</code> group.', 'error');
JFactory::getApplication()->enqueueMessage('Test 3: Message in <code>notice</code> group.', 'notice');
JFactory::getApplication()->enqueueMessage('Test 4: Message in <code>warning</code> group.', 'warning');
JFactory::getApplication()->enqueueMessage('Test 5: Message in <code>message</code> group (will be in the same queue as test 1).', 'message');
JFactory::getApplication()->enqueueMessage('Test 6: Message in <code>message without title</code> group.', 'message', array('showTitle' => false));
JFactory::getApplication()->enqueueMessage('Test 7: Message in <code>notice with custom title</code> group.', 'notice', array('customTitle' => 'My custom title'));
JFactory::getApplication()->enqueueMessage('Test 8: Message in <code>notice with custom title</code> group (message 2).', 'notice', array('customTitle' => 'My custom title'));
```
- Check if the result is something like this (please not that in beez3 there is no message title):
![image](https://cloud.githubusercontent.com/assets/9630530/14258865/4dda811e-fa9c-11e5-9281-30e21e8f2681.png)

An effort was made to preserve B/C with existent layout overwrites. You should also test this patch in a custom template that has a message layout override (in `/templates/<template-name>/html/layouts/joomla/system/message.php`) (or use the one of in isis without this PR changes).

Note that before this patch this messages would look like this:
![image](https://cloud.githubusercontent.com/assets/9630530/14259583/746a66f2-fa9f-11e5-98ba-4ef20d83a801.png)

#### Observations


@mbabker i had to change `JDocumentRendererHtmlMessage` and `enqueueMessage` (normal and legacy) to make this work with the new options, please check if everything is fine with this changes.

Any suggestions and/or improvements are welcomed.
